internal/git/cli.go

```diff
@@ -424,9 +454,27 @@ func (g *GitCLI) RemoteAdd(name string, url string) error {
 	return g.run("remote", "add", name, url)
 }
 
+func (g *GitCLI) Merge(repositoryURL, commitSha, workingDir string) (string, error) {
+	mergeBinary, err := exec.LookPath("merge")
```

Comment:
this seems like a very high risk of conflicts executable name. I would much rather have a hardcoded path with an env var that can override to support local testing. We will, in effect, be exposing a `merge` command to all steps in all jobs and we are then relying on our wrapper putting the binary on the tail of the PATH and hoping there is nothing in-between or that the user does not modify the path to add their own `merge` by appending rather than prepending.

In short, this name will work for the checkout action as we control the container and the action, but this name is too generic. If you want to pick it up from the PATH I would suggest `cloudbees-local-git-cache-helper` or `cloudbees-git-pr-merge-backfill` as the executable name as that is much less likely to conflict with anything in a user's step container... but much better is to define a location and name as part of the wrapper contract and check that location for the merge back-fill helper while allowing the fixed path to be overridden by an env var to make local testing tractable

Reply:
Renamed the binary to `cloudbees-git-pr-merge-backfill`